test/saithrift/tests/sail3.py

```diff
@@ -48,6 +48,9 @@ def runTest(self):
         nhop1 = sai_thrift_create_nhop(self.client, addr_family, ip_addr1, rif_id1)
         sai_thrift_create_route(self.client, vr_id, addr_family, ip_addr1_subnet, ip_mask1, rif_id1)
 
+        # Waiting the end of configuration ...
+        wait_till_configuration_will_end()
```

Comment:
the server side is running the saiserver, it only returns when the actually sai function call is returned.  maybe there is delay between your sai function call is return and the silicon emulator is populated with the entries.

can you make the delay a tunable parameter and pass via the command line and make the default parameter to 0 so that other platform does need the delay?